File: src/main/java/com/speak/score/dto/SpeechScoreResult.java
package com.speak.score.dto;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class SpeechScoreResult {
    private Double overallScore;
    private Double pronunciationScore;
    private Double fluencyScore;
    private Double completenessScore;
    private Double accuracyScore;
    private List<ErrorWord> errorWords;
    private boolean success;
    private String errorMessage;

    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class ErrorWord {
        private String word;
        private Double score;
        private String errorType;
        private Integer startIndex;
        private Integer endIndex;
    }
}

File: speak_score_flutter/lib/models/speech_score_result.dart
class SpeechScoreResult {
  final double? overallScore;
  final double? pronunciationScore;
  final double? fluencyScore;
  final double? completenessScore;
  final double? accuracyScore;
  final List<ErrorWord>? errorWords;
  final bool? success;
  final String? errorMessage;

  SpeechScoreResult({
    this.overallScore,
    this.pronunciationScore,
    this.fluencyScore,
    this.completenessScore,
    this.accuracyScore,
    this.errorWords,
    this.success,
    this.errorMessage,
  });

  factory SpeechScoreResult.fromJson(Map<String, dynamic> json) {
    final errorWordsList = json['errorWords'] as List?;
    return SpeechScoreResult(
      overallScore: (json['overallScore'] as num?)?.toDouble(),
      pronunciationScore: (json['pronunciationScore'] as num?)?.toDouble(),
      fluencyScore: (json['fluencyScore'] as num?)?.toDouble(),
      completenessScore: (json['completenessScore'] as num?)?.toDouble(),
      accuracyScore: (json['accuracyScore'] as num?)?.toDouble(),
      errorWords: errorWordsList
          ?.map((e) => ErrorWord.fromJson(e as Map<String, dynamic>))
          .toList(),
      success: json['success'] as bool?,
      errorMessage: json['errorMessage'] as String?,
    );
  }
}

class ErrorWord {
  final String? word;
  final double? score;
  final String? errorType;
  final int? startIndex;
  final int? endIndex;

  ErrorWord({
    this.word,
    this.score,
    this.errorType,
    this.startIndex,
    this.endIndex,
  });

  factory ErrorWord.fromJson(Map<String, dynamic> json) => ErrorWord(
        word: json['word'] as String?,
        score: (json['score'] as num?)?.toDouble(),
        errorType: json['errorType'] as String?,
        startIndex: json['startIndex'] as int?,
        endIndex: json['endIndex'] as int?,
      );
}
